sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+        dynModelId(idauto),
+        macroConnectionId(macroConnection) {}
+
+    /**
+     * @brief Equality operator
+     * @param other the other macro connect to compare to
+     * @returns true if they are equal, false if not
+     */
+    bool operator==(const MacroConnect& other) const;
+
+    /**
+     * @brief Non-equality operator
+     * @param other the other macro connect to compare to
+     * @returns true if they are not equal, false if not
+     */
+    bool operator!=(const MacroConnect& other) const;
+
+    DynModelDefinition::DynModelId dynModelId;                       ///< automaton id
+    DynModelDefinition::MacroConnection::MacroId macroConnectionId;  ///< macro connection id
+  };
+
+  /**
+   * @brief Hash structure for macro connec
+   */
+  struct MacroConnectHash {
+    /**
+     * @brief Operator to hash a macro connect
+     * @returns unique hash
+     */
+    std::size_t operator()(const MacroConnect& connect) const noexcept;
+  };
+
+ private:
+  /**
+   * @brief Determines if a library is loadable in current environement
+   * @param lib library name
+   * @returns true if library could be loaded, false if not
+   */
+  static bool doesLibraryExist(const std::string& lib);
+
+  /**
+   * @brief Computes library path for library name
+   *
+   * Determines the library path with Dynawo API first then using DFL environement
+   *
+   * @param lib the library name
+   * @returns the filepath of the library, or nullopt if not found
+   */
+  static boost::optional<boost::filesystem::path> computeLibPath(const std::string& lib);
+
+ private:
+  /// @brief Extract models from configuration before processing the nodes
+  void extractDynModels();
+
+  /**
+   * @brief Process single association from configuratio
+   *
+   * @param automaton the config automaton definition
+   * @param macro the macro connection connecting to the singleassociation
+   * @param singleassoc the single association config element to process
+   */
+  void dispatchAutomatonSingle(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton, const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                               const inputs::AssemblingXmlDocument::SingleAssociation& singleassoc);
+
+  /**
+   * @brief Process multi association from configuratio
+   *
+   * @param automaton the config automaton definition
+   * @param macro the macro connection connecting to the singleassociation
+   * @param multiassoc the multi association config element to process
+   */
+  void dispatchAutomatonMulti(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton, const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                              const inputs::AssemblingXmlDocument::MultipleAssociation& multiassoc);
+
+  /**
+   * @brief Process node in case of automaton bus connection
+   * @param node node to process
+   */
+  void processDynModelBusConnection(const NodePtr& node);
```

Comment:
I would propose addMacroConnectionToDynamicAutomaton (same idea for others)

Reply:
The idea of the function is to process the connection for current node so I would keek this name

Reply:
"process" is really generic

Reply:
That's the idea